### PR TITLE
feat: v8 12.8

### PR DESCRIPTION
**Pull request by devsnek**
includes exciting new features we can use:
- https://github.com/denoland/rusty_v8/pull/1542
- https://github.com/denoland/rusty_v8/pull/1543
- CPED in 12.8

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM!

Two asks:
- Could you update the PR/commit message with links to rusty_v8 PRs with "exciting new features" for reference?
- Can we hold off merging this PR until https://github.com/denoland/deno_core/pull/838 lands and we cut another release of `deno_core` (CC @nathanwhit)? Otherwise, we won't be able to land any other changes in Deno for 2-3 weeks, until we're a week away from next Deno minor release.

File: core/modules/map.rs [hunk 1]
```diff
@@ -777,6 +777,10 @@ impl ModuleMap {
       return Err(v8::Global::new(tc_scope, module.get_exception()));
     }
 
+    if module.get_status() != v8::ModuleStatus::Uninstantiated {
+      return Ok(());
+    }
```
Comment (bartlomieju) [inline core/modules/map.rs:782]: Actually this seems fishy - do we now have a failing test without this one?

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: core/modules/map.rs (hunk 1 above)
Comment (nathanwhit) [reply core/modules/map.rs:782]: This test fails `tests/testdata/run/088_dynamic_import_already_evaluating.ts`.

I looked into it yesterday (copied from slack):

> I think the issue is that we are calling instantiate_module multiple times if you dynamically import the same module multiple times – that never made sense but they added a check in Module::Instantiate to assert that the module wasn't already in the Instantiating state.

So pretty much yes it's incorrect, but it was always incorrect, this just gets things passing again and avoids the check failure.

Comment (bartlomieju) [reply core/modules/map.rs:782]: That makes sense, I think what you're talking about is a "linking phase"; before now, we could just instantiate any module and count on V8 to resolve them in the right order, but I guess that is no longer true. Thanks for the explanation. @devsnek could you add a comment around this line with this info?

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**